### DISCUSSION TITLE
Angular: .emit is not a function

### DISCUSSION THREAD

**Discussion by lovega** [storybookjs/storybook#16507] [Help]
Hello, 
 
I have created stories from a library of angular components, but in all I have the same error with the Output events.

![image](https://user-images.githubusercontent.com/40565772/139256646-f98df574-31b6-45cb-b7cc-b96fd82c4592.png)
![image](https://user-images.githubusercontent.com/40565772/139256688-5bc7e929-a7af-4616-a678-44684b0ef196.png)

The actions (changedTab) do not appear in the Addons panel and when clicking, the same error always appears: "ERROR TypeError: this.changedTab.emit is not a function"

![image](https://user-images.githubusercontent.com/40565772/139256025-1f33611e-dabd-4f7c-9183-33bba1b338ad.png)
![image](https://user-images.githubusercontent.com/40565772/139256199-0339c196-a80d-4cbe-a11f-2830d768ca8d.png)

The component works correctly importing in any angular application, the error is in the storybook.

How can I solve it?

Thanks!!

**Comment by samvloeberghs** [comment]
Hi @lovega, did you come to a solution in the end?

**Comment by lovega** [comment]
Hi! 
Yes, I finally found a solution. The `emit` must be included before `action`

`  changedTab: { emit: action('changedTab') },`
